pkg/querier/queryrange/log_result_cache.go

```diff
@@ -274,6 +273,45 @@ func (l *logResultCache) handleHit(ctx context.Context, cacheKey string, cachedR
 	return result, nil
 }
 
+// extractLokiResponse extracts response with interval [start, end)
+func extractLokiResponse(start, end time.Time, r *LokiResponse) *LokiResponse {
+	extractedResp := LokiResponse{
+		Status:     r.Status,
+		Direction:  r.Direction,
+		Limit:      r.Limit,
+		Version:    r.Version,
+		ErrorType:  r.ErrorType,
+		Error:      r.Error,
+		Statistics: r.Statistics,
+		Data: LokiData{
+			ResultType: r.Data.ResultType,
+			Result:     []logproto.Stream{},
+		},
+	}
+	for _, stream := range r.Data.Result {
+		if stream.Entries[0].Timestamp.After(end) || stream.Entries[len(stream.Entries)-1].Timestamp.Before(start) {
```

Comment:
The condition one above uses `ts.Nano()` where as here we use `time.After` for comparison. Can we make both same?

Reply:
I went with different ways to compare the time for convenience and readability.
It is hard to do `<=` or `>=` with `time.After` and `time.Before` while keeping it simple to read.
If I do the same here, then it would make the code a little inefficient due to having to do the conversion for each log line. I think it would be good to keep it as is, but please feel free to push back if you feel otherwise.